### PR TITLE
Add word2vec distribute model based on Fleet API

### PR DESCRIPTION
**Pull request by MrChengmo**
In this commit, following features was included:

-  Add skip-gram word2vec  distribute model based on Fleet API.

-  Support py_reader data read function, using sync mode between PSERVER and TRAINER.

-  Support dataset data read function, using async mode between PSEVER and TRAINER. 

-  Support prediction after training by first trainer node.

### CODE REVIEW

File: examples/distribute_ctr/argument.py [hunk 1]
```diff
@@ -24,133 +24,55 @@ def params_args(args=None):
...
-        default=10,
-        help="The number of passes to train (default: 10)")
-    params.add_argument(
-        '--sparse_feature_dim',
-        type=int,
-        default=1000001,
-        help='sparse feature hashing space for index processing')
+    params.add_argument('--embedding_size', type=int, default=10, help="The size for embedding layer (default:10)")
+    params.add_argument('--num_passes', type=int, default=10, help="The number of passes to train (default: 10)")
+    params.add_argument('--sparse_feature_dim', type=int, default=1000001,
```
Comment (seiriosPlus) [inline examples/distribute_ctr/argument.py]: 这些参数都是有用且有效的么？

File: examples/distribute_ctr/argument.py [hunk 2]
```diff
@@ -24,133 +24,55 @@ def params_args(args=None):
...
+                        help='sparse feature hashing space for index processing')
     params.add_argument('--dense_feature_dim', type=int, default=13)
-    params.add_argument(
-        '--no_split_var',
-        action='store_true',
-        default=False,
-        help='Whether split variables into blocks when update_method is pserver')
+    params.add_argument('--no_split_var', action='store_true', default=False,
```
Comment (seiriosPlus) [inline examples/distribute_ctr/argument.py]: 删除

File: examples/distribute_ctr/argument.py [hunk 3]
```diff
@@ -24,133 +24,55 @@ def params_args(args=None):
...
+                        help='The path for model to store (default: 127.0.0.1:6000)')
+    params.add_argument('-i', "--current_id", type=int, default=0, help="Specifies the number of the current role")
+    params.add_argument("--trainers", type=int, default=1,
+                        help="Specify the number of nodes participating in the training")
     params.add_argument("--is_first_trainer", type=bool, default=False)
     params.add_argument("--pserver_ip", type=str, default="127.0.0.1")
     params.add_argument("--pserver_endpoints", type=list, default=[])
     params.add_argument("--pserver_ports", type=str, default="36001")
-    params.add_argument("--sync_mode", type=bool, default=True)
+    params.add_argument("--sync_mode", type=bool, default=False)
```
Comment (seiriosPlus) [inline examples/distribute_ctr/argument.py]: sync_mode/async_mode/half_async_mode是互斥的.换一种写法

File: examples/distribute_ctr/argument.py [hunk 4]
```diff
@@ -24,133 +24,55 @@ def params_args(args=None):
...
+    params.add_argument("--half_async_mode", type=bool, default=False)
     params.add_argument("--async_mode", type=bool, default=False)
-    params.add_argument("--cpu_num", type=int, default=1)
+    params.add_argument("--cpu_num", type=int, default=2)
     params.add_argument("--use_cuda", type=bool, default=False)
```
Comment (seiriosPlus) [inline examples/distribute_ctr/argument.py]: 删除

File: examples/distribute_ctr/local_cluster.sh [hunk 5]
```diff
@@ -64,7 +66,8 @@ then
     for((i=0;i<$PADDLE_TRAINERS;i++))
     do
         echo "PADDLE WILL START Trainer "$i
-        PADDLE_TRAINER_ID=$i 
-	python -u model.py --is_${train_method}_train=True --is_local_cluster=True &> ./log/trainer.$i.log &
+        PADDLE_TRAINER_ID=$i
+	python -u model.py --is_${train_method}_train=True --is_local_cluster=True --${sync_mode}_mode=True &> ./log/trainer.$i.log &
```
Comment (seiriosPlus) [inline examples/distribute_ctr/local_cluster.sh]: 换写法

Comment (MrChengmo) [reply examples/distribute_ctr/local_cluster.sh]: Code has been modified as required

**Pull request merged**